lldb/source/Plugins/Language/Swift/SwiftUnsafeTypes.cpp

```diff
@@ -74,50 +114,108 @@ bool SwiftUnsafeBufferPointer::Update() {
 
-  ValueObjectSP some_value_sp = position_value_sp->GetChildAtIndex(0, true);
-  if (!some_value_sp || !some_value_sp->GetNumChildren())
+  if (!start_addr || start_addr == LLDB_INVALID_ADDRESS)
     return false;
 
-  CompilerType argument_type;
+  m_start_addr = start_addr;
 
-  if (CompilerType type = some_value_sp->GetCompilerType())
-    argument_type = SwiftASTContext::GetGenericArgumentType(type, 0);
+  return true;
+}
 
-  if (!argument_type.IsValid())
-    return false;
+class SwiftUnsafeRawBufferPointer final : public SwiftUnsafeType {
+public:
+  SwiftUnsafeRawBufferPointer(ValueObject &valobj);
+  bool Update() override;
 
-  m_elem_type = argument_type;
+private:
+  addr_t m_end_addr;
+};
 
-  ValueObjectSP pointer_value_sp = some_value_sp->GetChildAtIndex(0, true);
-  if (!pointer_value_sp)
+SwiftUnsafeRawBufferPointer::SwiftUnsafeRawBufferPointer(ValueObject &valobj)
+    : SwiftUnsafeType(valobj) {}
+
+bool SwiftUnsafeRawBufferPointer::Update() {
+  if (!m_valobj.GetNumChildren())
     return false;
 
-  addr_t addr = pointer_value_sp->GetValueAsUnsigned(LLDB_INVALID_ADDRESS);
+  // Here is the layout of Swift's UnsafeRaw[Mutable]BufferPointer.
+  // It's a view of the raw bytes of the pointee object. Each byte is viewed as
+  // a `UInt8` value independent of the type of values held in that memory.
+  //
+  // ▿ UnsafeRawBufferPointer
+  //   ▿ _position : Optional<UnsafeRawPointer<Int>>
+  //     ▿ some : UnsafeRawPointer<Int>
+  //       - pointerValue : Int
+  //   ▿ _end : Optional<UnsafeRawPointer<Int>>
+  //     ▿ some : UnsafeRawPointer<Int>
+  //       - pointerValue : Int
+  //
+  // The structure has 2 Optional UnsafePointer to the buffer's start address
+  // and end address. To access the pointer address, lldb unfolds every value
+  // object child until reaching `pointerValue`.
 
+  addr_t addr = GetAddress("_position");
   if (!addr || addr == LLDB_INVALID_ADDRESS)
     return false;
-
   m_start_addr = addr;
 
+  addr = GetAddress("_end");
+  if (!addr || addr == LLDB_INVALID_ADDRESS)
+    return false;
+  m_end_addr = addr;
+
+  if (!m_elem_type.IsValid()) {
+    auto ast_context = m_valobj.GetScratchSwiftASTContext();
+    if (!ast_context)
+      return false;
+
+    swift::TypeBase *swift_type =
+        ast_context->ReconstructType(ConstString("$ss5UInt8VD")); // Swift.UInt8
+    if (!swift_type)
+      return false;
+
+    CompilerType compiler_type = ast_context->GetCompilerType(swift_type);
+    if (!compiler_type.IsValid())
+      return false;
+
+    m_elem_type = compiler_type;
+  }
+
+  auto opt_type_size = m_elem_type.GetByteSize(m_valobj.GetTargetSP().get());
+
+  if (!opt_type_size)
+    return false;
+
+  m_count = (m_end_addr - m_start_addr) / *opt_type_size;
+
   return true;
 }
 
+std::unique_ptr<SwiftUnsafeType> SwiftUnsafeType::Create(ValueObject &valobj) {
+  llvm::StringRef valobj_type_name(valobj.GetTypeName().GetCString());
+  bool is_raw = valobj_type_name.contains("Raw");
```

Comment:
The formatter matching looks through typedefs.  So your valobj might have matched the formatter, but the top-level typename of this value object might be a typedef (and thus not contain "Raw").  The matching needs to also look through typedefs to the resolved type or you might get the wrong formatter.